### DISCUSSION TITLE
Need help on using private docker registry

### DISCUSSION THREAD

**Discussion by kk-bavenkatachalam** [linkerd/linkerd2#10023] [Help] [answered]
Hello,

Rather than using publicly hosted images, I would like to use private docker repository to install Linkerd on my cluster. 
I checked the [using a private docker repository](https://linkerd.io/2.12/tasks/using-a-private-docker-repository/) document and I have couple of questions regarding that.

1. As mentioned in the above document, the below command outputs 4 images used by linkerd.
```
linkerd install --ignore-cluster | grep image: | sed -e 's/^ *//' | sort | uniq
```

Output:
```
image:
image: cr.l5d.io/linkerd/controller:stable-2.12.3
image: cr.l5d.io/linkerd/policy-controller:stable-2.12.3
image: cr.l5d.io/linkerd/proxy-init:v2.0.0
image: cr.l5d.io/linkerd/proxy:stable-2.12.3
```
Importing these images to my docker registry is no big deal, but how do I pass the path of my private docker registry's image to the `linkerd install` command. 
I checked the [install](https://linkerd.io/2.12/reference/cli/install/) cli of linkerd but there is no option to override the default values.
Could someone please help me out here on how to override the default image paths while installing linkerd using linkerd cli?

2. On similar ground, I would also like to install `viz` extension. 
Could someone please help me out on this as well?

Any help is appreciated.
Thanks in advance!!

**Comment by alpeb** [comment, marked as answer]
For `linkerd install`, you can use the `--registry` flag:
```bash
$ linkerd install --registry foobar.com | grep image: | sed -e 's/^ *//' | sort | uniq
image:
image: foobar.com/controller:stable-2.12.3
image: foobar.com/policy-controller:stable-2.12.3
image: foobar.com/proxy-init:v2.0.0
image: foobar.com/proxy:stable-2.12.3
```
That flag is not available for `linkerd viz install` directly, but you can set the associated `values.yaml` entry:
```bash
$ linkerd viz install --set defaultRegistry=foobar.com | grep image: | sed -e 's/^ *//' | sort | uniq 
image: foobar.com/metrics-api:stable-2.12.3
image: foobar.com/tap:stable-2.12.3
image: foobar.com/web:stable-2.12.3
image: prom/prometheus:v2.30.3
```

**Reply by kk-bavenkatachalam** [reply]
@alpeb 
Thanks for the help!
